libnd4j/include/helpers/cpu/ConstantShapeHelper.cpp

```diff
@@ -28,318 +27,382 @@
 #include <helpers/logger.h>
 namespace sd {
 
+// Part 1: Constructor, Destructor, and Basic Instance Methods
+
 ConstantShapeHelper::~ConstantShapeHelper() {
 }
 
-
 ConstantShapeHelper::ConstantShapeHelper() {
-  _cache.resize(1);
-  for (int e = 0; e < 1; e++) {
-    SD_MAP_IMPL<ShapeDescriptor , ConstantShapeBuffer *> cache;
-    _cache[e] = cache;
-  }
 }
 
+ConstantShapeHelper& ConstantShapeHelper::getInstance() {
+ static ConstantShapeHelper instance;
+ return instance;
+}
 
-
-const sd::LongType * ConstantShapeHelper::emptyShapeInfoWithShape(const sd::DataType dataType,std::vector<sd::LongType> &shape) {
-  auto descriptor = ShapeBuilders::createShapeInfo(dataType,'c', shape, nullptr);
-  ArrayOptions::setPropertyBit(descriptor, ARRAY_EMPTY);
-  auto existing = createFromExisting(descriptor);
-  return existing;
+const sd::LongType * ConstantShapeHelper::emptyShapeInfoWithShape(const sd::DataType dataType, std::vector<sd::LongType> &shape) {
+ auto descriptor = ShapeBuilders::createShapeInfo(dataType,'c', shape, nullptr);
+ ArrayOptions::setPropertyBit(descriptor, ARRAY_EMPTY);
+ auto existing = createFromExisting(descriptor);
+ return existing;
 }
 
+} // namespace sd
+
+#endif/* ******************************************************************************
+*
+* This program and the accompanying materials are made available under the
+* terms of the Apache License, Version 2.0 which is available at
+* https://www.apache.org/licenses/LICENSE-2.0.
+*
+*  See the NOTICE file distributed with this work for additional
+*  information regarding copyright ownership.
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+* WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+* License for the specific language governing permissions and limitations
+* under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+******************************************************************************/
```

Comment:
Have you concatenated a couple files here? This file looks like a mess.

Reply:
Ah chunking issue. Fixed.